impeller/renderer/command_buffer.cc

```diff
@@ -51,6 +51,25 @@ bool CommandBuffer::SubmitCommandsAsync(
   return SubmitCommands(nullptr);
 }
 
+bool CommandBuffer::SubmitCommandsAsync(
```

Comment:
I haven't combed over this completely yet, but it seems a bit odd that we need to add a new submit method for this. Is there ever a case where we shouldn't emulate Metal's good behavior on the Vulkan backend?

Reply:
We're using SubmitCommandsAsync on macOS too. Fundamentally the old design was always 1. Encoder 2. Submit, but pushing that entire thing to a worker thread made the ownership a bit more complicated, having the command buffer take ownership of the render pass or blit pass makes things a bit clearer.

Reply:
Why do we have both `SubmitCommands` and `SubmitCommandsAsync`? From reading the docstrings, I can't distinguish why one should ever need to use `SubmitCommands`.

When is using `SubmitCommandsAsync` unsafe or undesirable?

Reply:
Oh at this point there is no reason to use SubmitCommands. Though in theory both should work, I can pare this down and just make SubmitCommandsAsync the only option (and rename it to SubmitCommands)

Reply:
actually there are two reasons we use this today:

1. Testing, sometimes we latch on the callback for the compute unittests.
2. The blit pass workloads that happen outside of the frame. Technically the way I've implemented submit commands async on vulkan assumes that there is some future step that will submit N command buffers, but there is no equivalent end of frame workload on image uploads.

So I don't know that we could entirely remove it, but we could re-arrange things to make it more obviously not the right thing to use for frame workload operations.